File: app/schemas/billing.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PlanResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str
    max_messages_per_month: int
    max_users: int
    max_whatsapp_instances: int
    max_ai_tokens_per_month: int


class SubscriptionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    company_id: int
    plan_id: int
    status: str
    starts_at: datetime
    ends_at: datetime | None

File: frontend/src/types/billing.ts
export type Plan = {
  id: number;
  code: string;
  name: string;
  max_messages_per_month: number;
  max_users: number;
  max_whatsapp_instances: number;
  max_ai_tokens_per_month: number;
};

export type Subscription = {
  id: number;
  company_id: number;
  plan_id: number;
  status: string;
  starts_at: string;
  ends_at: string | null;
};
